src/theory/quantifiers/sygus/synth_conjecture.cpp

```diff
@@ -346,14 +346,17 @@ bool SynthConjecture::doCheck(std::vector<Node>& lems)
   {
     // get the model value of the relevant terms from the master module
     std::vector<Node> enum_values;
-    bool fullModel = getEnumeratedValues(terms, enum_values);
+    bool activeIncomplete = false;
+    bool fullModel = getEnumeratedValues(terms, enum_values, activeIncomplete);
 
     // if the master requires a full model and the model is partial, we fail
     if (!d_master->allowPartialModel() && !fullModel)
     {
       // we retain the values in d_ev_active_gen_waiting
       Trace("cegqi-engine") << "...partial model, fail." << std::endl;
-      return true;
+      // if we are partial due to an active enumerator, we may still succeed
+      // on the next call
+      return !activeIncomplete;
```

Comment:
No, we need to call into the SygusModules before we have a full model. If we loop around getEnumeratedValues, it could take a very long time for an enumerator for some type to produce a new value (which is the point of this PR).  The complication that this addresses is that we need to ensure that the calls to SynthConjecture::check continue to be called within the call to SynthEngine::check.